tuple/test/engagement_test.cpp

```diff
@@ -0,0 +1,324 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <iostream>
+#include <iomanip>
+#include <set>
+#include <catch2/catch.hpp>
+#include <tuple_sketch.hpp>
+#include <tuple_union.hpp>
+#include <stdexcept>
+
+template<typename T>
+class max_value_policy {
+public:
+    max_value_policy(const T& initial_value): initial_value(initial_value) {}
+    T create() const { return initial_value; }
+    void update(T& summary, const T& update) const { summary = std::max(summary, update); }
+private:
+    T initial_value;
+};
+
+using max_float_update_tuple_sketch = datasketches::update_tuple_sketch<float, float, max_value_policy<float>>;
+
+template<typename T>
+class always_one_policy {
+public:
+    always_one_policy(): initial_value(1) {}
+    T create() const { return 1; }
+    void update(T& summary, const T& update) const { }
+private:
+    T initial_value;
+};
+using always_one_tuple_sketch = datasketches::update_tuple_sketch<int, int, always_one_policy<int>> ;
+
+template<typename T>
+class update_sum_value_policy {
+public:
+    //update_sum_value_policy(const T& initial_value): initial_value(0) {}
```

Comment:
ideally we'd remove comments like this when checking in a final version

Reply:
Comments removed.